src/flows/service_functions/communication.py

```diff
@@ -91,9 +92,15 @@ def send_message(
         }
     )
     if target_state is None:
-        cast(Any, receiver).msg(parsed, **kwargs)
+        if isinstance(receiver, MsgCapable):
+            receiver.msg(parsed, **kwargs)
+        elif hasattr(receiver, 'msg'):
+            receiver.msg(parsed, **kwargs)
     else:
-        cast(Any, target_state).msg(parsed, **kwargs)
+        if isinstance(target_state, MsgCapable):
+            target_state.msg(parsed, **kwargs)
+        elif hasattr(target_state, 'msg'):
+            target_state.msg(parsed, **kwargs)
```

Comment:
Neither of these feel necessary. We should get a state from the above calls and BaseState has a `msg` method. If it can't determine the types of receiver or target_state we should assert that they're instances of BaseState or its subclasses